packet_entity.go

```diff
@@ -27,16 +38,31 @@ func (pe *PacketEntity) Fetch(key string) (interface{}, bool) {
 	return v, ok
 }
 
+// A function that can handle a game event.
+type packetEntityHandler func(*PacketEntity, EntityEventType) error
+
+// Registers a packet entity event handler.
+func (p *Parser) OnPacketEntity(fn packetEntityHandler) {
+	p.packetEntityHandlers = append(p.packetEntityHandlers, fn)
+}
+
 // Internal callback for CSVCMsg_PacketEntities.
 func (p *Parser) onCSVCMsg_PacketEntities(m *dota.CSVCMsg_PacketEntities) error {
+	// Skip processing if we're configured not to.
 	if !p.ProcessPacketEntities {
 		return nil
 	}
 
 	_debugfl(5, "pTick=%d isDelta=%v deltaFrom=%d updatedEntries=%d maxEntries=%d baseline=%d updateBaseline=%v", p.Tick, m.GetIsDelta(), m.GetDeltaFrom(), m.GetUpdatedEntries(), m.GetMaxEntries(), m.GetBaseline(), m.GetUpdateBaseline())
 
+	// Skip processing full updates after the first. We'll process deltas instead.
```

Comment:
Yeah this is controversial. Thanks for bringing it up.

I'm not sure if this is acceptable in all cases. It's acceptable for all matches in the test suite. I tested locally by adding a panic if any of these packets (which would be omitted) made any KV changes, and they didn't.

It still may be reasonable to back out this change and eat the ~5% perf hit rather than risk a hard-to-diagnose bug in the future.